### PR TITLE
Added support for PointGrey Spinnaker API compatible devices

### PR DESCRIPTION
**Pull request by markasselin**

### CODE REVIEW

File: Modules/FindSpinnakerAPI.cmake [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    "../../PLTools/PointGrey/windows/spinnaker"
+    "../trunk/PLTools/PointGrey/windows/spinnaker"
+    "$ENV{PROGRAMFILES}/Point Grey Research/Spinnaker"
+    "C:/Program Files/Point Grey Research/Spinnaker"
+    "$ENV{${_x86env}}/Point Grey Research/Spinnaker"
+    "C:/Program Files (x86)/Point Grey Research/Spinnaker"
+    )
+ELSEIF(UNIX)
+  # Ubuntu Spinnaker API path hints
+
```
Comment (adamrankin) [inline Modules/FindSpinnakerAPI.cmake:52]: Do they have a linux SDK install?

Comment (adamrankin) [reply Modules/FindSpinnakerAPI.cmake:52]: Same `UNIX AND NOT MAC` comment

File: Modules/FindSpinnakerAPI.cmake [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  # get release codename
+  execute_process(COMMAND lsb_release -cs
```
Comment (adamrankin) [inline Modules/FindSpinnakerAPI.cmake:54]: command is not available on OSX

File: Modules/FindSpinnakerAPI.cmake [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    mark_as_advanced(SPINNAKER_API_LIBRARY_DIR)
+    set(SPINNAKER_API_BINARY_DIR /usr/lib)
+    mark_as_advanced(SPINNAKER_API_BINARY_DIR)
+
+    #Version
+    spinnaker_get_version("/usr/include/spinnaker/System.h")
+  ENDIF()
+ENDIF()
+
+include(FindPackageHandleStandardArgs)
```
Comment (adamrankin) [inline Modules/FindSpinnakerAPI.cmake:126]: Would you consider populating a CMake target with the correct details?

Windows example can be found here:
https://github.com/PlusToolkit/PlusBuild/blob/master/Modules/FindAgilentMD1.cmake#L60
Linux example here:
https://github.com/PlusToolkit/PlusBuild/blob/master/Modules/FindV4L2.cmake#L49

**Comment by markasselin** [comment]
@adamrankin Thanks for all the helpful comments! @Sunderlandkyl the Spinnaker video device is ready to integrate, maybe add it right after the new PLUS stable is frozen?

**Pull request merged**